src/Symfony/Component/VarDumper/Dumper/AbstractDumper.php

```diff
@@ -131,4 +173,40 @@ protected function echoLine($line, $depth, $indentPad)
             fwrite($this->outputStream, str_repeat($indentPad, $depth).$line."\n");
         }
     }
+
+    /**
+     * Converts a non-UTF-8 string to UTF-8.
+     *
+     * @param string $s The non-UTF-8 string to convert.
+     *
+     * @return string The string converted to UTF-8.
+     */
+    protected function utf8Encode($s)
+    {
+        if ('mbstring' === $this->charsetConverter) {
+            return mb_convert_encoding($s, 'UTF-8', mb_check_encoding($s, $this->charset) ? $this->charset : '8bit');
+        }
+        if ('iconv' === $this->charsetConverter) {
+            $valid = true;
+            set_error_handler(function () use (&$valid) {$valid = false;});
+            $c = iconv($this->charset, 'UTF-8', $s);
+            restore_error_handler();
+            if ($valid) {
+                return $c;
+            }
+        }
+
+        $s .= $s;
+        $len = strlen($s);
+
+        for ($i = $len >> 1, $j = 0; $i < $len; ++$i, ++$j) {
+            switch (true) {
+                case $s[$i] < "\x80": $s[$j] = $s[$i]; break;
+                case $s[$i] < "\xC0": $s[$j] = "\xC2"; $s[++$j] = $s[$i]; break;
```

Comment:
please format it on multiple line to make it more readable (seeing more easily what is the condition and what are the executed statements)